src/App.tsx

```diff
@@ -15,71 +15,89 @@ export const goods = [
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  selectGood = (good: string) => {
+    this.setState({
+      selectedGood: good,
+    });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  clearSelection = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    return (
+      <main className="section container">
+        {this.state.selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {this.state.selectedGood}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+            {' is selected'}
 
-        <tr data-cy="Good">
-          <td>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={this.clearSelection}
+            />
+          </h1>
+        ) : (
+          <h1 className="title">No goods selected</h1>
+        )}
+
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              const isSelected = good === this.state.selectedGood;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              return (
+                <tr
+                  key={good}
+                  data-cy="Good"
+                  className={isSelected ? 'has-background-success-light' : ''}
+                >
+                  <td>
+                    {isSelected ? (
+                      <button
+                        data-cy="RemoveButton"
+                        type="button"
+                        className="button is-info"
+                        onClick={this.clearSelection}
+                      >
+                        -
+                      </button>
+                    ) : (
+                      <button
```

Comment:
Each condition should start from the new line
```suggestion
                    {isSelected 
                      ? (
                      <button
                        data-cy="RemoveButton"
                        type="button"
                        className="button is-info"
                        onClick={this.clearSelection}
                      >
                        -
                      </button>
                    ) 
                      : (
                      <button
```